src/cloudai/_core/installables/base.py

```
@@ -0,0 +1,91 @@
+
+class InstallStatusResult:
+    """
+    Class representing the result of an installation, uninstallation, or status check.
+
+    Attributes
+        success (bool): Indicates whether the operation was successful.
+        message (str): A message providing additional information about the result.
+        details (Optional[Dict[str, str]]): A dictionary containing details about the result for each test template.
+    """
+
+    def __init__(self, success: bool, message: str = "", details: dict[Installable, Self] | None = None):
```

Ruff (0.15.12)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)